src/main/scala/io/qbeast/spark/QbeastTable.scala

```diff
@@ -65,6 +65,52 @@ class QbeastTable private (
       .map(_.getString(0))
   }
 
+  def getIndexMetrics(revisionID: Option[RevisionID] = None): IndexMetrics = {
+    val allCubeStatuses = qbeastSnapshot.loadLatestIndexStatus.cubesStatuses
+
+    val cubeCounts = allCubeStatuses.size
+    val depth = allCubeStatuses.map(_._1.depth).max
+    val row_count = allCubeStatuses.flatMap(_._2.files.map(_.elementCount)).sum
+    val dimensionCount = allCubeStatuses.toList.head._1.dimensionCount
+    val desiredCubeSize = qbeastSnapshot.loadLatestRevision.desiredCubeSize
+
+    val nonLeafStatuses =
+      allCubeStatuses.filter(_._1.children.exists(allCubeStatuses.contains)).values
+    val nonLeafCubeSizes = nonLeafStatuses.flatMap(_.files.map(_.size)).toSeq.sorted
```

Comment:
Bug found!
Here you are mapping to the file `size` in bytes instead of the `elementCount`

Reply:
Fixed!